dir.targets

```diff
@@ -56,4 +56,11 @@
     </SupplementalTestData>
   </ItemGroup>
 
+  <Target Name="ProducesPackageId" 
+          Returns="@(PackageIds)">
+    <ItemGroup>
+      <PackageIds Include="$(Id)" />
```

Comment:
Are there instances within the package .builds files where Id is set outside of a package context?  You may not know off hand, and you may just be forward thinking.   From my initial survey, this looked appropriate, but I can try to find something more explicitly tied to packages if there is a concern here.